src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -1297,10 +1297,25 @@ public void InternalOnDownstreamFinish(Exception cause)
         {
             try
             {
-                if (cause == null)
-                    throw new ArgumentException("Cancellation cause must not be null", nameof(cause));
                 if (_lastCancellationCause != null)
                     throw new ArgumentException("OnDownstreamFinish must not be called recursively", nameof(cause));
+                
+                // Some stages might propagate null exceptions due to improper Task continuation handling
+                // (see https://github.com/akkadotnet/Akka.Persistence.Sql/issues/498)
+                // This is a stop gap solution to make sure that Akka.Streams doesn't behave improperly
+                // until we can fix all of those
```

Comment:
Document the reason in the code for future developers

Reply:
LGTM